sycl/include/sycl/queue.hpp

```diff
@@ -95,6 +96,60 @@ class __SYCL_EXPORT SubmissionInfo {
 private:
   std::shared_ptr<SubmissionInfoImpl> impl = nullptr;
 };
+#endif
+
+namespace v1 {
```

Comment:
This way one particular data structure claims entire `sycl::detail::vN` namespace, so that we can't use the same approach for other APIs (that's the same reason why I think our `sycl::_v1` turned out to be rather useless). IMO, it would be better to have `detail::submission_info::_v1::SubmissionInfo` or something like that. Maybe make `submission_info` inline itself.